managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1482,18 +1482,17 @@ public synchronized void createComplete(int rc, final LedgerHandle lh, Object ct
             lastLedgerCreationFailureTimestamp = clock.millis();
         } else {
             log.info("[{}] Created new ledger {}", name, lh.getId());
-            ledgers.put(lh.getId(), LedgerInfo.newBuilder().setLedgerId(lh.getId()).setTimestamp(0).build());
-            currentLedger = lh;
-            currentLedgerEntries = 0;
-            currentLedgerSize = 0;
-
             final MetaStoreCallback<Void> cb = new MetaStoreCallback<Void>() {
                 @Override
                 public void operationComplete(Void v, Stat stat) {
                     if (log.isDebugEnabled()) {
                         log.debug("[{}] Updating of ledgers list after create complete. version={}", name, stat);
                     }
                     ledgersStat = stat;
+                    ledgers.put(lh.getId(), LedgerInfo.newBuilder().setLedgerId(lh.getId()).setTimestamp(0).build());
```

Comment:
@lordcheng10 Adding to the ledgers map cannot be done here, because that map is used when writing to the z-node (eg: `buildManagedLedgerInfo(Map<Long, LedgerInfo> ledgers)`.

Reply:
fixed. PTAL,thanks! @merlimat

Reply:
I still have concerns because in this method, if the ledger is not persisted to the metadata, it means the ledger cannot be used. But we put it in the ledger map; it is not persistent and may be used by other operations under multi-threaded conditions.

However, I'm not sure what it will affect. Just a minor concern. Please feel free to go on.

Reply:
Your concern is right, I will try to update the zookeeper metadata with a temporary ledgersTmp, and update the ledgers after the update is successful. @mattisonchao @merlimat

Reply:
Agree with @mattisonchao. This may lead to data loss.

Putting the newly created ledger into the `NavigableMap<Long, LedgerInfo> ledgers`, the new ledger can be seen immediately which means can write data into this ledger, and then if the meta-store operation failed, the ledger will be removed from both the `ledgers` and bookie, all data in this ledger are lost.

Reply:
fixed. @merlimat @hangc0276 @aloyszhang @mattisonchao PTAL,thanks!